src/app/api/eventVolunteer/[eventVolunteerId]/routes.ts

```diff
@@ -0,0 +1,34 @@
+import { NextRequest, NextResponse } from 'next/server';
+import { zObjectId } from '@/types/dataModel/base';
+import CMError, { CMErrorResponse, CMErrorType } from '@/utils/cmerror';
+import { zCheckInVolunteerRequest } from '@/types/dataModel/eventVolunteer';
+import { checkInVolunteerAction } from '@/server/actions/EventVolunteer';
+
+export async function PUT(
+  req: NextRequest,
+  { params }: { params: { eventVolunteerId: string; checkInTime: Date } }
+) {
+  try {
+    const idValidationResult = zObjectId.safeParse(params.eventVolunteerId);
+    if (!idValidationResult.success) {
+      return new CMError(
+        CMErrorType.BadValue,
+        'EventVolunteer Id'
+      ).toNextResponse();
+    }
+
+    const data = await req.json();
```

Comment:
The request will only contain the date, so you will need to pass the date and the id into your safeParse function below. i.e. instead of data, {_id: params.eventVolunteerId, checkInDate: data.checkInDate} or something like that.

Reply:
pass the same argument to checkInVolunteerAction